src/components/EmailLoginForm/EmailLoginForm.tsx

```diff
@@ -10,168 +10,234 @@ import IconButton from '@mui/material/IconButton';
 import { styled, Theme } from "@mui/material/styles";
 import VisibilityIcon from '@mui/icons-material/Visibility';
 import VisibilityOffIcon from '@mui/icons-material/VisibilityOff';
-import { signInWithEmailAndPassword } from 'firebase/auth';
 import Tooltip from '@mui/material/Tooltip';
-import useLoginForm from '../../hooks/useLoginForm'
-import { Controller, SubmitHandler } from "react-hook-form";
+import useLoginForm from '../../hooks/useLoginForm';
+import { FieldValues, SubmitHandler } from "react-hook-form";
+import { useNavigate } from 'react-router-dom';
+import { Container } from '@mui/material';
 
 const EmailLoginForm = () => {
-    const [showPassword, setShowPassword] = useState(false); // Estado para controlar se a senha está visível
-    const [buttonClicked, setButtonClicked] = useState(false); // Estado para controlar se o botão foi clicado
+    const [showPassword, setShowPassword] = useState(false);
+    const [buttonClicked, setButtonClicked] = useState(false);
     const [dialogOpen, setDialogOpen] = useState(false);
-    const { register, watch, control, handleSubmit, reset } = useLoginForm();
+    const [erroLogin, setErroLogin] = useState(false);
+    const navigate = useNavigate(); // Hook para navegação
+    const { register, watch, handleSubmit, reset } = useLoginForm();
+    const [ error, setError ] = useState<string | null>(null);
 
     const handleClickShowPassword = () => setShowPassword((show) => !show);
 
     const handleMouseDownPassword = (event: React.MouseEvent<HTMLButtonElement>) => {
         event.preventDefault();
-      };
-
-    const openDialog = () => {
-        setDialogOpen(true);
     };
 
-    const closeDialog = () => {
-      setDialogOpen(false);
-    };
+    const openDialog = () => setDialogOpen(true);
+
+    const closeDialog = () => setDialogOpen(false);
 
     const watchEmail = watch('email');
     const watchSenha = watch('senha');
     const watchEmailRedefinicaoSenha = watch('emailRedefinicaoSenha');
 
-    const submitForm: SubmitHandler<{ email: string | undefined, senha: string | undefined, emailRedefinicaoSenha: string }> = async (data) => {
+    useEffect(() => {
+        if (!dialogOpen) {
+            reset({ emailRedefinicaoSenha: '' });
+        }
+    }, [dialogOpen, reset]);
+
+    const submitForm: SubmitHandler<FieldValues> = async (data) => {
         try {
-            await signInWithEmailAndPassword(auth, data.email!, data.senha!);
+          await signInWithEmailAndPassword(auth, data.email!, data.senha!).catch(
+            () => {
+              setErroLogin(true);
+              setError('Erro ao fazer login');
+              throw new Error('Erro ao fazer login');
+            }
+          );
+          navigate('/home');
         } catch (error) {
+            setErroLogin(true);
+            setError('Erro ao fazer login');
             throw new Error('Erro ao fazer login');
```

Comment:
Erro relançado pela promise é tratado nesse block catch, por exemplo, se o código for auth/invalid-email mostrar isso no helperText do input de email